C-Plus-Plus/search/bilinear_search.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+We are given an array a[], check whether the value exists or not,
+if exits print the index(1 based indexing) else print -1.
+
+Input:
+first line containes size of array.
+Next line containes n space seperated intgeres.
+Next line contains a value that needed to be searched.
+
+Sample Input 1: 
+5
+5 4 3 2 1
+1
+
+Sample Output 1:
+5
+
+Sample Input 2: 
+5
+5 4 3 2 1
+7
+
+Sample Output 2:
+-1
+
+Bilinear approach:
+Let A be the linear array of n elements. The algorithm searches for element 'key'. 
+Let 'index' represent the location of the element 'key' in the array. 
+The algorithm returns the values index=-1 if the element 'key' is not present in the array 
+and value index if the element 'key' is present in the array.
+
+Language: C++17
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+#define MAX 100005
+
+//bilenear search algorithm
+int bilinear(vector<int>&a,int n,int key)
+{
+    int front=0,back=n-1,index=-1;
+    //terminating condition
+    while(front<=back)
+    {
+        if(a[front]==key)
+            index=front+1;
+        if(a[back]==key)
+            index=back+1;
+        front++;
+        back--;
+    }
+    return index;

```

Comment:
and here you can just return -1 as the key isn't present if this lines gets executed.